src/main/resources/swagger/api-docs.yaml

```diff
@@ -3633,14 +3665,66 @@ components:
           example: my-managed-resource-group
       description: specifies a request for a pet managed identity
     User:
+      type: object
       required:
+        - id
         - email
-      type: object
+        - enabled
+        - createdAt
+        - updatedAt
       properties:
+        id:
+          type: string
+          description: User's Id
+        googleSubjectId:
+          type: string
+          description: User's Google subject Id
         email:
           type: string
           description: User's email address
+          format: email
+        azureB2CId:
+          type: string
+          description: User's Azure B2C Id
+        enabled:
+          type: boolean
+          description: Whether or not the user is enabled
+        acceptedTosVersion:
+          type: string
+          description: The version of terms of service the user accepted
+        createdAt:
+          type: string
+          format: date-time
+          description: User's time of creation
+        registeredAt:
+          type: string
+          format: date-time
+          description: User's time of registration
+        updatedAt:
+          type: string
+          format: date-time
+          description: User's time of last update
       description: specification of a User
+    UpdateUserRequest:
+      type: object
+      properties:
+        googleSubjectId:
+          type: string
+          description: User's Google subject Id
+        email:
+          type: string
+          description: User's email address
+          format: email
+        azureB2CId:
+          type: string
+          description: User's Azure B2C Id
+        enabled:
+          type: boolean
+          description: Whether or not the user is enabled
+        acceptedTosVersion:
```

Comment:
The accepted TOS version should only be updated via the TOS endpoints.

Reply:
This is another thing I wanted to talk about. I do agree that I think only a user can accept their tos (although I think it might be valuable to edit the version if for some reason there is a data discrepancy), but I think eventually we would want to do away with the specific endpoints for updating one field of a user. For now, I can take this out of the swagger doc, would you also like me to remove it from the UpdateUserRequest object in code (right now it is not doing anything) or just leave that as it is?

Reply:
The Update model should probably only contain update-able fields, since our API definition should reflect the functionality of the endpoint. As for the B2C ID, I we need to do a lot of hard thinking about if that should be update-able. I don't think we've gotten any user requests or support tickets involving updating the B2C ID for a user. We do know that there's a need to update emails and Google Subject IDs, so that's probably what we should focus on and provide for now.

Reply:
But, to another point, a set of CRUD endpoints should probably use the same object for requests and responses. So, if we were to have just a `SamUser` object with all the user fields, and use that as both the request and response, then we could say that the only fields that are update-able are the email and Google Subject ID.

Reply:
I think that we should not be using the same request and response objects since it makes it difficult to know what is required in a request vs what you know is in the response (that is what made updating the `PolicyIdentifiers `complex in the first place). So for example, in the `GET` user endpoint, we can say that the email is a required field. But in the `PATCH` user endpoint, email is not required. This could be done in the `POST` user endpoint by marking certain required field as read only since you need a complete object to make this request. So I still think that there should be a `UpdateUserRequest` object, but I will update it to only have the fields we currently support updating.